File: back_end/src/main/java/org/example/cmpe202_final/service/course/GradeService.java
package org.example.cmpe202_final.service.course;

import lombok.AllArgsConstructor;
import org.example.cmpe202_final.model.course.Grade;
import org.example.cmpe202_final.model.course.GradeWithStudentName;
import org.example.cmpe202_final.repository.grades.CustomGradeRepository;
import org.example.cmpe202_final.repository.grades.GradeRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@AllArgsConstructor
@Service
public class GradeService {

    private final CustomGradeRepository customGradeRepository;
    private final GradeRepository gradeRepository;

    public List<GradeWithStudentName> getGradesWithStudentNamesByAssignmentId(String assignmentId, String userId){
        return customGradeRepository.getGradesWithStudentNamesByAssignmentId(assignmentId, userId);
    }

    public Grade updateGrade(GradeWithStudentName gradeWithStudentName) {
        return customGradeRepository.updateGrade(gradeWithStudentName);
    }

}

File: front_end/app/src/main/java/com/example/studentportal/grades/service/GradeService.kt

import com.example.studentportal.grades.usecase.model.GradeUseCaseModel
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface GradeService {
    @GET("/grades/assignment/{assignmentId}")
    fun fetchGradesByAssignment(
        @Path("assignmentId") assignmentId: String,
        @Query("userId") userId: String
    ): Call<List<GradeUseCaseModel>>

    @PUT("/grades/update")
    fun updateGrade(
        @Body gradeWithStudentName: GradeUseCaseModel
    ): Call<GradeUseCaseModel>
}
